src/renderer/components/modules/SettingsWindow.tsx

```diff
@@ -22,12 +22,53 @@ const updateSettings = async (req: { apiKey: string }): Promise<string> => {
   }
 };
 
+const getSettings = async (): Promise<{
+  settings: StringMap;
+  errMsg: string;
+}> => {
+  return window.electronAPI
+    .get_settings()
+    .then((resp) => {
+      console.log("settings: ", resp);
+      if ( 'errMsg' in resp ) {
+        return {
+          settings: {},
+          errMsg: resp.errMsg,
+        };
+      }
+      return {
+        settings: resp,
+        errMsg: "",
+      };
+    })
+    .catch((e) => {
+      console.log("fail to get settings:", e);
+      return {
+        settings: {},
+        errMsg: "fail to get settings",
+      };
+    });
+};
+
 const SettingsWindow = (props: {
   setOpenSettings: (b: boolean) => void;
   callGetUserEvents: () => void;
 }) => {
   const [loading, setLoading] = useState(false);
   const [errorMsg, setErrorMsg] = useState("");
+  const [openai_api_key, setApiKey] = useState("");
```

Comment:
same problem of variable naming.

Reply:
A quick question, are there situations that snack case is used in JS?